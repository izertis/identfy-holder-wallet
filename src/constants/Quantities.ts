export const THREE_SECONDS_SNACK_DURATION = 3000;
export const INT_MAX_VALUE = 2147483648;
const QUANTITIES = {
  THREE_SECONDS_SNACK_DURATION,
  INT_MAX_VALUE,
};
export default QUANTITIES;
