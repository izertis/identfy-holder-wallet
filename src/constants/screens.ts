export const SCREEN = {
	SplashScreen: 'SplashScreen',
	OnBoarding: 'OnBoarding',
	Register: 'Register',
	Login: 'Login',
	Recovery: 'Recovery',
	SecurityPhrase: 'SecurityPhrase',
	SecurityPhraseConfirm: 'SecurityPhraseConfirm',
	DerivationPath: 'DerivationPath',
	NetworkAuth: 'NetworkAuth',
	DidList: 'DidList',
	QrReader: 'QrReader',
	Credentials: 'Credentials',
	Presentations: 'Presentations',
	Historical: 'Historical',
	Help: 'Help',
} as const
