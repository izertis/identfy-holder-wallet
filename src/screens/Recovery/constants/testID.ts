export const testID = {
  FIRST_PIN_INPUT: "first-pin-input",
  SECOND_PIN_INPUT: "second-pin-input",
  REGISTER_ERROR_MESSAGE: "register-error-message",
}
