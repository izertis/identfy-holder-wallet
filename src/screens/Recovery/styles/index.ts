import defaultStyles from "./defaultStyles"
import proyectSyles from "./proyectStyles"

const RecoveryStyled = {
  ...defaultStyles,
  ...proyectSyles
}

export default RecoveryStyled
