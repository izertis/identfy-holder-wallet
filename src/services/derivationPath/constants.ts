import { settingType } from "./types"

const IDENTFY_Z_KEY = '1037171'

export enum derivationPathFormats {
  MAIN_IDENTITY = "mZRSSSSSW",
  NETWORK_DID = "mMTN",
  INTERACTING_DID = "mMTNB"
}

export const derivationPathSeetingsDefault: settingType = {
  Z: IDENTFY_Z_KEY,
  R: "random",
  SSSSS: "random",
  W: "random",
  B: "random"
}

export const KEYCHILDREN = {
  // When adding a derivation path for keychildren, we need to have a input from the other party so we can get the identifier for that party (e.g. CashBank, MadridUniversity, etc), and automate it so every time we interact with the other party we get the identifier for that party.
}



