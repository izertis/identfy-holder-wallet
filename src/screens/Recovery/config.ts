import i18next from '../../../i18n.config'
import { SCREEN } from "../../constants/screens"
import { navigationScreensConfigType } from "../../types/navigation"
import es from "./i18n/es"
import Recovery from '.'
import { ColorKeys, getThemeColor } from '../../constants/Colors'

const ScreenName = SCREEN.Recovery

i18next.addResourceBundle("es", ScreenName, es)

const RecoveryConfig: navigationScreensConfigType = {
  name: ScreenName,
  component: Recovery,
  options: { headerShown: false },
}

export default RecoveryConfig
