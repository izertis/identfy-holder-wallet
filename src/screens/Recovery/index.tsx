import React, { useContext, useEffect, useState } from 'react'
import { useTranslation } from 'react-i18next'
import { KeyboardAvoidingView } from 'react-native'
import { RootStackScreenProps } from '../../../types'
import { SCREEN } from '../../constants/screens'
import { MessageContext } from '../../context/UserMessage.context'
import { changePin, checkSecurityPhrase } from '../../utils/keychain'
import { testID } from './constants/testID'
import RecoveryI18nKeys from './i18n/keys'
import RecoveryStyled from './styles'

const Recovery = ({ navigation }: RootStackScreenProps<'Recovery'>) => {
	const { t } = useTranslation(SCREEN.Recovery)

	const { showMessage } = useContext(MessageContext)
	const [securityPhrase, setSecurityPhrase] = useState('')
	const [firstPin, setFirstPin] = useState('')
	const [isPinFilled, setIsPinFilled] = useState(false)
	const [secondPin, setSecondPin] = useState('')
	const [isLoading, setIsLoading] = useState(false)

	useEffect(() => {
		setIsPinFilled(arePinEquals())
	}, [firstPin, secondPin])

	const setNewPin = async (securityPhrase: string, pin: string) => {
		setIsLoading(true)
		const isCorrectPhrase = await checkSecurityPhrase(securityPhrase.trim())
		if (isCorrectPhrase && isPinLengthValid(secondPin)) {
			changePinAndClearForm(pin)
		} else if (!isCorrectPhrase) {
			showPhraseError()
		} else {
			showPinError()
		}
	}

	const changePinAndClearForm = async (pin: string) => {
		const isChanged = await changePin(pin)
		if (isChanged) {
			clearForm()
			const content = t(RecoveryI18nKeys.SUCCESS)
			showMessage({ content, type: 'success' })
			setIsLoading(false)
			navigation.navigate('Login')
		}
	}

	const clearForm = (): void => {
		setSecurityPhrase('')
		setFirstPin('')
		setSecondPin('')
		setIsLoading(false)
	}

	const arePinEquals = () => {
		return firstPin !== '' && secondPin !== '' && firstPin === secondPin
	}

	const showPhraseError = (): void => {
		const content = t(RecoveryI18nKeys.ERROR_PHRASE)
		showMessage({ content, type: 'error' })
		setIsLoading(false)
	}

	const showPinError = (): void => {
		const content = t(RecoveryI18nKeys.ERROR_PIN_LENGTH)
		showMessage({ content, type: 'error' })
		setIsLoading(false)
	}

	const isPinLengthValid = (pin: string) => {
		return pin.length > 3
	}

	const renderErrorMessage = () => {
		if (
			firstPin === '' ||
			secondPin === '' ||
			arePinEquals()
		) {
			return ''
		}
		return t(RecoveryI18nKeys.ERROR_PIN_EQUALS)
	}

	return (
		<KeyboardAvoidingView>
			<RecoveryStyled.MainContainer>
				<RecoveryStyled.RecoveryButton onPress={() => navigation.pop()}>
					{t(RecoveryI18nKeys.GO_BACK)}
				</RecoveryStyled.RecoveryButton>
				<RecoveryStyled.Title testID='title'>{t(RecoveryI18nKeys.TITLE)}</RecoveryStyled.Title>
				<RecoveryStyled.Subtitle>{t(RecoveryI18nKeys.SUBTITLE)}</RecoveryStyled.Subtitle>

				<RecoveryStyled.InputsContainer>
					<RecoveryStyled.InputStyled
						value={securityPhrase}
						multiline
						onChangeText={(value: string) => setSecurityPhrase(value)}
						placeholder={t(RecoveryI18nKeys.SECURITY_KEY).toString()}
						returnKeyType='next'
					/>
					<RecoveryStyled.InputStyled
						testID={testID.FIRST_PIN_INPUT}
						value={firstPin}
						keyboardType='numeric'
						secureTextEntry
						onChangeText={(value: string) => setFirstPin(value.replace(/[^0-9]/g, ''))}
						placeholder={t(RecoveryI18nKeys.PIN) || ''}
						returnKeyType='next'
					/>
					<RecoveryStyled.InputStyled
						testID={testID.SECOND_PIN_INPUT}
						value={secondPin}
						keyboardType='numeric'
						secureTextEntry
						onChangeText={(value: string) => setSecondPin(value.replace(/[^0-9]/g, ''))}
						placeholder={t(RecoveryI18nKeys.REPIN) || ''}
						returnKeyType={'done'}
					/>
					{renderErrorMessage() != '' && (
						<RecoveryStyled.ErrorText testID={testID.REGISTER_ERROR_MESSAGE}>
							{renderErrorMessage()}
						</RecoveryStyled.ErrorText>
					)}
				</RecoveryStyled.InputsContainer>

				<RecoveryStyled.ContainerBottom>
					{isLoading ? <RecoveryStyled.SplashActivityIndicator size={35} /> : null}
					<RecoveryStyled.Button
						disabled={!securityPhrase || !isPinFilled || isLoading}
						onPress={() => {
							setNewPin(securityPhrase, secondPin)
						}}
					>
						{isLoading ? t('LOADING') : t('BUTTON')}
					</RecoveryStyled.Button>
				</RecoveryStyled.ContainerBottom>
			</RecoveryStyled.MainContainer>
		</KeyboardAvoidingView>
	)
}

export default Recovery
