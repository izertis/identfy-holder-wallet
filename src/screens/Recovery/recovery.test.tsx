import { NativeStackNavigationProp } from '@react-navigation/native-stack'
import { fireEvent, render } from '@testing-library/react-native'
import Recovery from '.'
import { RootStackParamList } from '../../../types'
import { testID } from './constants/testID'

type ScreenNavigationProp = NativeStackNavigationProp<RootStackParamList, 'Recovery'>

const mockedNavigation: ScreenNavigationProp | any = {
	navigate: jest.fn(),
	route: jest.fn(),
	translate: jest.fn(),
}

describe('Recovery screen', () => {
	it('should render recovery screen', () => {
		const result = render(<Recovery {...mockedNavigation} />)

		expect(result.toJSON()).toMatchSnapshot()
	})
	it('Should renders items correctly', () => {
		const { getAllByRole, getByTestId } = render(<Recovery {...mockedNavigation} />)

		const title = getByTestId('title')
		const confirmBtn = getAllByRole('button')
		const goBack = getAllByRole('button')

		expect(title).toBeDefined()
		expect(confirmBtn).toBeDefined()
		expect(goBack).toBeDefined()
	})

	it('renders an error message if the pins are not equal', () => {
		const { getByTestId, queryAllByTestId } = render(<Recovery {...mockedNavigation} />)

		const firstPinInput = getByTestId(testID.FIRST_PIN_INPUT)
		const secondPinInput = getByTestId(testID.SECOND_PIN_INPUT)

		fireEvent.changeText(firstPinInput, '1234')
		fireEvent.changeText(secondPinInput, '5678')

		let messageErrorText = queryAllByTestId(testID.REGISTER_ERROR_MESSAGE)
		expect(messageErrorText).toHaveLength(1)

		fireEvent.changeText(secondPinInput, '1234')
		messageErrorText = queryAllByTestId(testID.REGISTER_ERROR_MESSAGE)
		expect(messageErrorText).toHaveLength(0)
	})
})
