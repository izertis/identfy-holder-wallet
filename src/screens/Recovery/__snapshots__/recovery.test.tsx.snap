// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recovery screen should render recovery screen 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "marginLeft": 25,
          "marginRight": 25,
          "marginTop": 15,
          "paddingBottom": 35,
        },
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "marginBottom": "6%",
          "marginLeft": "-1%",
          "marginTop": "5%",
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "width": "22%",
        }
      }
      testID="button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "transparent",
            "borderRadius": 20,
            "borderStyle": "solid",
            "borderWidth": 0,
            "flex": undefined,
            "minWidth": 64,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container"
      >
        <View
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 20,
              },
            ]
          }
          testID="button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                },
                undefined,
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "marginLeft": 12,
                    "marginRight": -4,
                  },
                  Object {
                    "marginLeft": 16,
                    "marginRight": -16,
                  },
                  Object {
                    "marginLeft": 12,
                    "marginRight": -8,
                  },
                ]
              }
              testID="button-icon-container"
            >
              <Text />
            </View>
            <Text
              numberOfLines={1}
              selectable={false}
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "fontFamily": "System",
                      "fontSize": 14,
                      "fontWeight": "500",
                      "letterSpacing": 0.1,
                      "lineHeight": 20,
                    },
                    Array [
                      Object {
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      false,
                      Object {
                        "marginHorizontal": 16,
                      },
                      undefined,
                      false,
                      Object {
                        "color": "rgba(103, 80, 164, 1)",
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Object {
                        "color": "#eb5b5b",
                      },
                    ],
                  ],
                ]
              }
              testID="button-text"
            >
              GO_BACK
            </Text>
          </View>
        </View>
      </View>
    </View>
    <Text
      bold={true}
      className="sc-bcXHqe sc-hLBbgP sc-iBYQkv kFXtiY RwpaV"
      fontFamily="Montserrat"
      style={
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          Object {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
      testID="title"
    >
      TITLE
    </Text>
    <Text
      className="sc-bcXHqe sc-dkrFOg sc-ftTHYK laZgtI bNRHGr eAnpSx"
      style={
        Array [
          Object {
            "textAlign": "left",
          },
          Object {
            "color": "rgba(28, 27, 31, 1)",
            "fontFamily": "System",
            "fontWeight": "400",
            "letterSpacing": 0,
          },
          Object {
            "writingDirection": "ltr",
          },
          undefined,
        ]
      }
    >
      SUBTITLE
    </Text>
    <RCTScrollView
      style={
        Array [
          Object {},
        ]
      }
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              undefined,
            ]
          }
        >
          <View
            style={Object {}}
          >
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderColor": "#363d58",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 0,
                    },
                    Object {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  className="sc-gKPRtg sc-pyfCe kwDacX fTbXGf"
                  containerStyle={
                    Object {
                      "paddingHorizontal": 0,
                    }
                  }
                  cursorColor="#363d58"
                  editable={true}
                  errorStyle={
                    Object {
                      "color": "#eb5b5b",
                    }
                  }
                  inputContainerStyle={
                    Object {
                      "marginTop": 10,
                    }
                  }
                  inputStyle={
                    Object {
                      "fontSize": 16,
                    }
                  }
                  maxFontSizeMultiplier={1.5}
                  multiline={true}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="SECURITY_KEY"
                  placeholderTextColor="#aaa"
                  returnKeyType="next"
                  selectionColor="#eb5b5b"
                  style={
                    Array [
                      Object {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      Object {},
                      Object {
                        "paddingBottom": 16,
                        "paddingTop": 16,
                      },
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "top",
                      },
                      false,
                      Array [
                        Object {},
                      ],
                      Object {
                        "fontFamily": "Muli",
                        "lineHeight": 25,
                      },
                    ]
                  }
                  testID="text-input-outlined"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              undefined,
            ]
          }
        >
          <View
            style={Object {}}
          >
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderColor": "#363d58",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 0,
                    },
                    Object {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  className="sc-gKPRtg sc-pyfCe kwDacX fTbXGf"
                  containerStyle={
                    Object {
                      "paddingHorizontal": 0,
                    }
                  }
                  cursorColor="#363d58"
                  editable={true}
                  errorStyle={
                    Object {
                      "color": "#eb5b5b",
                    }
                  }
                  inputContainerStyle={
                    Object {
                      "marginTop": 10,
                    }
                  }
                  inputStyle={
                    Object {
                      "fontSize": 16,
                    }
                  }
                  keyboardType="numeric"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="PIN"
                  placeholderTextColor="#aaa"
                  returnKeyType="next"
                  secureTextEntry={true}
                  selectionColor="#eb5b5b"
                  style={
                    Array [
                      Object {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      Object {
                        "height": 48,
                      },
                      Object {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      Object {
                        "marginRight": 56,
                        "paddingRight": 0,
                      },
                      Object {
                        "fontFamily": "Muli",
                        "lineHeight": 25,
                      },
                    ]
                  }
                  testID="first-pin-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "height": 24,
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": 24,
                    },
                    Object {
                      "right": 16,
                      "top": 19,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 20,
                      "height": 40,
                      "margin": 0,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                      "width": 40,
                    }
                  }
                  testID="right-icon-adornment-container-outer-layer"
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "rgba(121, 116, 126, 1)",
                        "borderRadius": 20,
                        "borderWidth": 0,
                        "elevation": 0,
                        "flex": 1,
                        "overflow": "hidden",
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0,
                        "shadowRadius": 0,
                      }
                    }
                    testID="right-icon-adornment-container"
                  >
                    <View
                      accessibilityComponentType="button"
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessibilityTraits="button"
                      accessible={true}
                      centered={true}
                      collapsable={false}
                      focusable={true}
                      hitSlop={
                        Object {
                          "bottom": 6,
                          "left": 6,
                          "right": 6,
                          "top": 6,
                        }
                      }
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexGrow": 1,
                              "justifyContent": "center",
                            },
                            Object {
                              "borderRadius": 20,
                            },
                          ],
                        ]
                      }
                      testID="right-icon-adornment"
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#fff",
              },
              undefined,
            ]
          }
        >
          <View
            style={Object {}}
          >
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "rgba(255, 251, 254, 1)",
                    "borderColor": "#363d58",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                  undefined,
                ]
              }
              testID="text-input-outline"
            />
            <View>
              <View
                style={
                  Array [
                    Object {
                      "paddingBottom": 0,
                    },
                    Object {
                      "minHeight": 56,
                      "paddingTop": 8,
                    },
                  ]
                }
              >
                <TextInput
                  className="sc-gKPRtg sc-pyfCe kwDacX fTbXGf"
                  containerStyle={
                    Object {
                      "paddingHorizontal": 0,
                    }
                  }
                  cursorColor="#363d58"
                  editable={true}
                  errorStyle={
                    Object {
                      "color": "#eb5b5b",
                    }
                  }
                  inputContainerStyle={
                    Object {
                      "marginTop": 10,
                    }
                  }
                  inputStyle={
                    Object {
                      "fontSize": 16,
                    }
                  }
                  keyboardType="numeric"
                  maxFontSizeMultiplier={1.5}
                  multiline={false}
                  onBlur={[Function]}
                  onChangeText={[Function]}
                  onFocus={[Function]}
                  placeholder="REPIN"
                  placeholderTextColor="#aaa"
                  returnKeyType="done"
                  secureTextEntry={true}
                  selectionColor="#eb5b5b"
                  style={
                    Array [
                      Object {
                        "margin": 0,
                        "zIndex": 1,
                      },
                      Object {
                        "height": 48,
                      },
                      Object {
                        "paddingBottom": 0,
                        "paddingTop": 0,
                      },
                      Object {
                        "color": "rgba(28, 27, 31, 1)",
                        "fontFamily": "System",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": 0.15,
                        "lineHeight": undefined,
                        "paddingHorizontal": 16,
                        "textAlign": "left",
                        "textAlignVertical": "center",
                      },
                      false,
                      Object {
                        "marginRight": 56,
                        "paddingRight": 0,
                      },
                      Object {
                        "fontFamily": "Muli",
                        "lineHeight": 25,
                      },
                    ]
                  }
                  testID="second-pin-input"
                  underlineColorAndroid="transparent"
                  value=""
                />
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "height": 24,
                      "justifyContent": "center",
                      "position": "absolute",
                      "width": 24,
                    },
                    Object {
                      "right": 16,
                      "top": 19,
                    },
                  ]
                }
              >
                <View
                  collapsable={false}
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "borderRadius": 20,
                      "height": 40,
                      "margin": 0,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                      "width": 40,
                    }
                  }
                  testID="right-icon-adornment-container-outer-layer"
                >
                  <View
                    collapsable={false}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderColor": "rgba(121, 116, 126, 1)",
                        "borderRadius": 20,
                        "borderWidth": 0,
                        "elevation": 0,
                        "flex": 1,
                        "overflow": "hidden",
                        "shadowColor": "#000",
                        "shadowOffset": Object {
                          "height": 0,
                          "width": 0,
                        },
                        "shadowOpacity": 0,
                        "shadowRadius": 0,
                      }
                    }
                    testID="right-icon-adornment-container"
                  >
                    <View
                      accessibilityComponentType="button"
                      accessibilityRole="button"
                      accessibilityState={
                        Object {
                          "disabled": false,
                        }
                      }
                      accessibilityTraits="button"
                      accessible={true}
                      centered={true}
                      collapsable={false}
                      focusable={true}
                      hitSlop={
                        Object {
                          "bottom": 6,
                          "left": 6,
                          "right": 6,
                          "top": 6,
                        }
                      }
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Array [
                            Object {
                              "alignItems": "center",
                              "flexGrow": 1,
                              "justifyContent": "center",
                            },
                            Object {
                              "borderRadius": 20,
                            },
                          ],
                        ]
                      }
                      testID="right-icon-adornment"
                    >
                      <Text />
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "flex-end",
            "marginBottom": 40,
            "marginLeft": "auto",
            "marginRight": "auto",
            "marginTop": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 20,
              "borderStyle": "solid",
              "borderWidth": 0,
              "flex": undefined,
              "minWidth": 64,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="button-container"
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 20,
                },
              ]
            }
            testID="button"
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Object {
                    "backgroundColor": "#ccc",
                    "borderRadius": 15,
                    "fontFamily": "Muli",
                    "height": 39,
                    "width": 224,
                  },
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "textAlign": "left",
                    },
                    Object {
                      "color": "rgba(28, 27, 31, 1)",
                      "writingDirection": "ltr",
                    },
                    Array [
                      Object {
                        "fontFamily": "System",
                        "fontSize": 14,
                        "fontWeight": "500",
                        "letterSpacing": 0.1,
                        "lineHeight": 20,
                      },
                      Array [
                        Object {
                          "marginHorizontal": 16,
                          "marginVertical": 9,
                          "textAlign": "center",
                        },
                        false,
                        Object {
                          "marginHorizontal": 12,
                        },
                        undefined,
                        false,
                        Object {
                          "color": "rgba(28, 27, 31, 0.38)",
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        Object {
                          "color": "#fff",
                          "fontFamily": "Muli",
                          "letterSpacing": 0.57,
                        },
                      ],
                    ],
                  ]
                }
                testID="button-text"
              >
                BUTTON
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
